modules/fargate/fargate-service/main.tf

```diff
@@ -0,0 +1,292 @@
+  retention_in_days = var.log_retention_in_days
+  tags              = var.tags
+}
+
+# ------------------------------------------------------------------------------
+# IAM - Task execution role, needed to pull ECR images etc.
+# ------------------------------------------------------------------------------
+resource "aws_iam_role" "execution" {
+  name               = "${var.name_prefix}-task-execution-role"
+  assume_role_policy = data.aws_iam_policy_document.task_assume.json
+}
+
+resource "aws_iam_role_policy" "task_execution" {
+  name   = "${var.name_prefix}-task-execution"
+  role   = aws_iam_role.execution.id
+  policy = data.aws_iam_policy_document.task_execution_permissions.json
+}
+
+resource "aws_iam_role_policy" "read_repository_credentials" {
+  count  = length(var.repository_credentials) != 0 ? 1 : 0
+  name   = "${var.name_prefix}-read-repository-credentials"
+  role   = aws_iam_role.execution.id
+  policy = data.aws_iam_policy_document.read_repository_credentials.json
+}
+
+# ------------------------------------------------------------------------------
+# IAM - Task role, basic. Users of the module will append policies to this role
+# when they use the module. S3, Dynamo permissions etc etc.
+# ------------------------------------------------------------------------------
+resource "aws_iam_role" "task" {
+  name               = "${var.name_prefix}-task-role"
+  assume_role_policy = data.aws_iam_policy_document.task_assume.json
+}
+
+resource "aws_iam_role_policy" "log_agent" {
+  name   = "${var.name_prefix}-log-permissions"
+  role   = aws_iam_role.task.id
+  policy = data.aws_iam_policy_document.task_permissions.json
+}
+
+# ------------------------------------------------------------------------------
+# Security groups
+# ------------------------------------------------------------------------------
+resource "aws_security_group" "ecs_service" {
+  vpc_id      = var.vpc_id
+  name        = "${var.name_prefix}-ecs-service-sg"
+  description = "Fargate service security group"
+  tags = merge(
+    var.tags,
+    {
+      Name = "${var.name_prefix}-sg"
+    },
+  )
+}
+
+resource "aws_security_group_rule" "egress_service" {
+  security_group_id = aws_security_group.ecs_service.id
+  type              = "egress"
+  protocol          = "-1"
+  from_port         = 0
+  to_port           = 0
+  cidr_blocks       = ["0.0.0.0/0"]
+  ipv6_cidr_blocks  = ["::/0"]
+}
+
+resource "aws_security_group_rule" "ingress_service" {
+  security_group_id = aws_security_group.ecs_service.id
+  type              = "ingress"
+  protocol          = "-1"
+  from_port         = var.task_container_port
+  to_port           = var.task_container_port
+  cidr_blocks       = var.allowed_subnets
+}
+
+# ------------------------------------------------------------------------------
+# LB Target group
+# ------------------------------------------------------------------------------
+resource "aws_lb_target_group" "task" {
```

Comment:
It seems like the `aws_lb_target_group`, `aws_ecs_task_definition` and `aws_ecs_service` resources here could be replaced by a call to the `service` submodule of this repository (after doing slight modifications). This would avoid duplication of code, and ensure that any changes/improvements benefit both regular ECS and Fargate usecases.